src/App.tsx

```diff
@@ -14,11 +14,80 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">
+          {selectedGoods.length === 0
+            ? 'No goods selected'
+            : (
+              <>
+                {`Selected Goods: ${selectedGoods.join(', ')}`}
+                <button
+                  type="button"
+                  className="App__clearButton"
+                  onClick={() => {
+                    this.setState({ selectedGoods: [] });
+                  }}
+                >
+                  Clear
+                </button>
+              </>
+            )}
+        </h1>
+        <ul className="App__list">
+          {goodsFromServer.map(good => {
+            return selectedGoods.includes(good)
+              ? (
+                <li className="App__listItem App__listItem--selected">
+                  {good}
+                  {' - '}
+                  <button
+                    type="button"
+                    className="
+                      App__listItemButton
+                      App__listItemButton--toRemove
+                    "
+                    onClick={() => {
+                      selectedGoods.splice(selectedGoods.indexOf(good), 1);
+                      this.setState({ selectedGoods });
+                    }}
```

Comment:
Here you are mutating state.selectedGoods - it's prohibited.
Use `filter` and move this logic to a separate method.

The same case with Select button, fix it there, as well